Fix `build.client` and `build.server` resolve behaviour (#11916)
withastro/astro: .changeset/chilly-terms-know.md

---
'astro': major
---

> Updates how the `build.client` and `build.server` option values get resolved. As currently documented, the option values will now correctly resolve relative to the `outDir` option. So if `outDir` is set to `./dist/nested/`, then by default:

Reviewer: ```suggestion
Updates how the `build.client` and `build.server` option values get resolved to match existing documentation. With this fix, the option values will now correctly resolve relative to the `outDir` option. So if `outDir` is set to `./dist/nested/`, then by default:
```

> - `build.client` will resolve to `<root>/dist/nested/client/`
> - `build.server` will resolve to `<root>/dist/nested/server/`
> 
> Previously the values were incorrectly resolved:
> 
> - `build.client` was resolved to `<root>/dist/nested/dist/client/`
> - `build.server` was resolved to `<root>/dist/nested/dist/server/`

Reviewer 2: ```suggestion
- `build.client` will resolve to `<root>/<outDir>/nested/client/`
- `build.server` will resolve to `<root>/<outDir>/nested/server/`

Previously the values were incorrectly resolved:

- `build.client` was resolved to `<root>/<outDir>/nested/<outDir>/client/`
- `build.server` was resolved to `<root>/<outDir>/nested/<outDir>/server/`
```

> If you were relying on the previous build paths, make sure that it's adapted to the new build paths.

Reviewer: ```suggestion
If you were relying on the previous build paths, make sure that your project code is updated to the new build paths.
```
